src/aposteriori/dnn/data_processing/tools.py

```diff
@@ -1,14 +1,68 @@
+import copy
 import h5py
+import sys
 import typing as t
 import warnings
-import copy
 
 from ampal.amino_acids import standard_amino_acids
 from collections import Counter
 from pathlib import Path
 from random import shuffle
 
-from aposteriori.dnn.config import UNCOMMON_RESIDUE_DICT
+from aposteriori.data_prep.create_frame_data_set import DatasetMetadata
+from aposteriori.dnn.config import UNCOMMON_RESIDUE_DICT, MAKE_FRAME_DATASET_VER
+
+
+def extract_metadata_from_dataset(frame_dataset: Path) -> DatasetMetadata:
+    """
+    Retrieves the metadata of the dataset and does a sanity check of the version.
+    If the dataset version is not compatible with aposteriori, the training process will stop.
+
+    Parameters
+    ----------
+    frame_dataset: Path
+        Path to the .h5 dataset with the following structure.
+        └─[pdb_code] Contains a number of subgroups, one for each chain.
+          └─[chain_id] Contains a number of subgroups, one for each residue.
+            └─[residue_id] voxels_per_side^3 array of ints, representing element number.
+              └─.attrs['label'] Three-letter code for the residue.
+              └─.attrs['encoded_residue'] One-hot encoding of the residue.
+        └─.attrs['make_frame_dataset_ver']: str - Version used to produce the dataset.
+        └─.attrs['frame_dims']: t.Tuple[int, int, int, int] - Dimentsions of the frame.
+        └─.attrs['atom_encoder']: t.List[str] - Lables used for the encoding (eg, ["C", "N", "O"]).
+        └─.attrs['encode_cb']: bool - Whether a Cb atom was added at the avg position of (-0.741287356, -0.53937931, -1.224287356).
+        └─.attrs['atom_filter_fn']: str - Function used to filter the atoms in the frame.
+        └─.attrs['residue_encoder']: t.List[str] - Ordered list of residues corresponding to the encoding used.
+        └─.attrs['frame_edge_length']: float - Length of the frame in Angstroms (A)
+
+    Returns
+    -------
+    dataset_metadata: DatasetMetadata of the dataset with the following parameters:
+        make_frame_dataset_ver: str
+        frame_dims: t.Tuple[int, int, int, int]
+        atom_encoder: t.List[str]
+        encode_cb: bool
+        atom_filter_fn: str
+        residue_encoder: t.List[str]
+        frame_edge_length: float
+
+    """
+    with h5py.File(frame_dataset, "r") as dataset_file:
+        meta_dict = dict(dataset_file.attrs.items())
+        dataset_metadata = DatasetMetadata.import_metadata_dict(meta_dict)
+
+    # Extract version metadata:
+    dataset_ver_num = dataset_metadata.make_frame_dataset_ver.strip(".")[0]
+    aposteriori_ver_num = MAKE_FRAME_DATASET_VER.strip(".")[0]
```

Comment:
I think this is meant to be:

```suggestion
    dataset_ver_num = dataset_metadata.make_frame_dataset_ver.split(".")[0]
    aposteriori_ver_num = MAKE_FRAME_DATASET_VER.split(".")[0]
```

Your current code works with the current version, but not in the way that I think you intended. `"0.1.0".strip(".")` returns `"0.1.0"`, then you're taking the first element, with is the first character of the string, which is "0". But is the version was `"10.0.2"`, then this code would give a version of "1". I'm assuming you wanted split to produce `["0", "1", "0"]` or `["10", "0", "2"]` then you take the major version off the front giving "0" or "10". Or am I miss reading it?